lib/kms.js

```diff
@@ -23,62 +22,59 @@ exports.createKeystore = async ({
   const config = {
     sequence: 0,
     controller: capabilityAgent.id,
-    // TODO: prefer using just controller when the invoker and delegator will be
-    //       the same
-    invoker: capabilityAgent.id,
-    delegator: capabilityAgent.id
   };
   if(referenceId) {
     config.referenceId = referenceId;
   }
-  const {httpsAgent} = brHttpsAgent;
   return KmsClient.createKeystore({
     url: `${kmsBaseUrl}/keystores`,
     config,
     httpsAgent
   });
 };
 
-// FIXME: this only works with an integrated KMS right now and that needs to be
-//        fixed in the future
-exports.updateKeystoreController = async ({id, controller} = {}) => {
-  const {httpsAgent} = brHttpsAgent;
-  const config = await KmsClient.getKeystore({id, httpsAgent});
-
-  const {invoker} = config;
-  if(!Array.isArray(invoker)) {
-    config.invoker = [invoker];
+exports.updateKeystoreController = async ({
+  controller, keystoreAgent, keystoreConfig
+} = {}) => {
+  // if the keystore config is not supplied, retrieve it now
+  if(!keystoreConfig) {
+    // FIXME: GET ID
+    // keystoreConfig = await KmsClient.getKeystore({id, httpsAgent});
+  } else {
+    keystoreConfig = clone(keystoreConfig);
   }
 
   // update config
-  config.sequence++;
-  const {controller: oldController} = config;
-  config.controller = controller;
-  // replace existing controller with new one
-  config.invoker = config.invoker.map(
-    x => x === oldController ? controller : x);
-  if(Array.isArray(config.delegator)) {
-    config.delegator = config.delegator.map(
-      x => x === oldController ? controller : x);
-  } else if(config.delegator === oldController) {
-    config.delegator = controller;
+  keystoreConfig.sequence++;
+  keystoreConfig.controller = controller;
+
+  // FIXME: keep referenceId?
+  delete keystoreConfig.referenceId;
+
+  let err;
+  let result;
+  try {
+    result = await keystoreAgent.updateConfig({config: keystoreConfig});
+  } catch(e) {
+    err = e;
   }
-  const updated = await keystores.update({config});
-  if(!updated) {
+  if(!result?.success) {
```

Comment:
Shouldn't there just be an exception thrown from `keystores.update` if the update didn't work?